src/expandable-section/expandable-section-header.tsx

```diff
@@ -190,12 +188,19 @@ const ExpandableHeaderTextWrapper = ({
         </InternalHeader>
       ) : (
         <>
-          <HeadingTag
-            className={clsx(styles['header-wrapper'], headingTagListeners && styles['click-target'])}
-            {...headingTagListeners}
-          >
-            {headerButton}
-          </HeadingTag>
+          {variant === 'compact' ? (
+            <div className={clsx(styles['header-wrapper'], styles['header-wrapper-compact'])}>
+              <HeadingTag className={styles['click-target']}>{headerButton}</HeadingTag>
```

Comment:
Because we only want the key to be clickable and not the value, whereas the default and footer variants today make the entire header wrapper clickable, although this is something we want to change in the future (see doc: `7SZJAvbhbznX`). When that happens, this code will be a bit simpler. For now, I have refactored this is a bit further, so hopefully it's a bit more clear.